diff -urN nemiver-0.6.3/configure.ac nemiver-0.6.3.new/configure.ac
--- nemiver-0.6.3/configure.ac	2008-09-27 21:32:55.000000000 +0200
+++ nemiver-0.6.3.new/configure.ac	2008-10-09 19:52:06.000000000 +0200
@@ -70,11 +70,7 @@
 AC_LANG_CPLUSPLUS
 AC_LANG_COMPILER_REQUIRE
 
-AC_PATH_PROG(GDB_PROG,gdb)
-
-if test x$GDB_PROG = x ; then
-    AC_MSG_ERROR([didn't find the gdb debugger. please, install it.])
-fi
+GDB_PROG="/usr/bin/gdb"
 AC_SUBST(GDB_PROG)
 
 ENABLE_IDEBUGGER_ONLY=no
