--- nemiver-0.9.6/src/common/libnemivercommon.pc.in.orig	2013-01-21 13:05:35.000000000 +0100
+++ nemiver-0.9.6/src/common/libnemivercommon.pc.in	2022-04-11 19:16:11.539192501 +0200
@@ -6,6 +6,6 @@ includedir=@NEMIVER_INCLUDE_DIR@
 Name: libnemivercommon
 Description: Common infrastructure for the Nemiver debugger and library
 Version: @VERSION@
-Requires: libxml-2.0 >= @LIBXML2_VERSION@, glibmm-2.4 >= @LIBGLIBMM_VERSION@, gnome-vfs-2.0 >= @LIBGNOMEVFS_VERSION@, gmodule-2.0 >= @LIBGMODULE_VERSION@, gthread-2.0 >= @LIBGTHREAD_VERSION@, libgtop-2.0 >= @LIBGTOP_VERSION@
+Requires: libxml-2.0 >= @LIBXML2_VERSION@, glibmm-2.4 >= @LIBGLIBMM_VERSION@, gmodule-2.0 >= @LIBGMODULE_VERSION@, gthread-2.0 >= @LIBGTHREAD_VERSION@, libgtop-2.0 >= @LIBGTOP_VERSION@
 Libs: -L${libdir} -lnemivercommon
 Cflags: -I${includedir}
